graspologic/layouts/__main__.py

```diff
@@ -0,0 +1,312 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import argparse
+import logging
+from pathlib import Path
+import sys
+from typing import Any, Dict, List
+
+import networkx as nx
+
+from . import auto, NodePosition, render
+from .colors import categorical_colors
+
+
+def _graph_from_file(
+    path: str,
+    skip_header: bool = False,
+) -> nx.Graph:
+    graph = nx.Graph()
+    with open(path, "r") as edge_io:
+        if skip_header is True:
+            next(edge_io)
+        for line in edge_io:
+            source, target, weight = line.strip().split(",")
+            weight = float(weight)
+            if graph.has_edge(source, target):
+                weight += graph[source][target]["weight"]
+            graph.add_edge(source, target, weight=weight)
+    return graph
+
+
+def _ensure_output_dir(path: str):
+    Path(path).parent.mkdir(parents=True, exist_ok=True)
+
+
+def _location(path: str, positions: List[NodePosition], colors: Dict[Any, str]):
+    with open(path, "w") as node_positions_out:
+        print("id,x,y,size,community,color", file=node_positions_out)
+        for position in positions:
+            print(
+                f"{position.node_id},{position.x},{position.y},{position.size},"
+                f"{position.community},{colors[position.node_id]}",
+                file=node_positions_out,
+            )
+
+
+def _output(
+    arguments: argparse.Namespace, graph: nx.Graph, positions: List[NodePosition]
+):
+    partitions = {position.node_id: position.community for position in positions}
+    color_map = categorical_colors(partitions)
+    if arguments.image_file is not None:
+        render.save_graph(arguments.image_file, graph, positions, node_colors=color_map)
+    if arguments.location_file is not None:
+        _location(arguments.location_file, positions, color_map)
+
+
+def _tsne(arguments: argparse.Namespace):
+    valid_args(arguments)
+    graph = _graph_from_file(arguments.edge_list, arguments.skip_header)
+    graph, positions = auto.layout_tsne(
+        graph, perplexity=30, n_iter=1000, max_edges=arguments.max_edges
+    )
```

Comment:
This seems to be missing the ability to remove the overlap from nodes. I think we need to have that added back in there.